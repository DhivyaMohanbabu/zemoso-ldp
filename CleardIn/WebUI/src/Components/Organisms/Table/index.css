.ag-header-cell{
  padding-top: 15px;
  display: inline;
}

.ag-react-container{
    width:100%;
}