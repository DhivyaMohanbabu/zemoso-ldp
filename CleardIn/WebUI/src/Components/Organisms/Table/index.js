/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import { AgGridColumn, AgGridReact } from 'ag-grid-react';
import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-balham.css';

import CheckBoxSharpIcon from '@material-ui/icons/CheckBoxSharp';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import { Checkbox, Typography } from '@material-ui/core';
import TitleIcon from '@material-ui/icons/TitleSharp';
import { makeStyles } from '@material-ui/core/styles';

import ColoredOption from '../../Atoms/ColoredOption';
import TableHeader from '../../Molecules/TableHeader';
import RowAdder from '../../Molecules/RowAdder';

import './index.css';

const useStyles = makeStyles({
  checkbox: {
    padding: '0px',
    paddingLeft: '0.5',
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'row',

  },
  textStyle: {
    paddingTop: '8px',
  },
  indexStyle: {
    paddingTop: '3px',
  },
});

function Table({ rows }) {
  const classes = useStyles();
  const [checkedIndexes, setCheckedIndexes] = useState([]);
  const [hoveredIndex, sethoveredIndex] = useState(-1);

  const relationshipToColorMapping = {
    customer: '#fdd570',
    vendor: '#daefa9',
    partner: '#b0f1f0',
  };

  const RelationshipRenderer = ({ value }) => (
    <ColoredOption
      label={value}
      color={relationshipToColorMapping[value]}
    />
  );

  const getRowHeight = (params) => (params.node.group ? 72 : 45);

  const onClickHandler = (event) => {
    const rowIndex = event.target.id;
    setCheckedIndexes((prevArray) => {
      // eslint-disable-next-line no-param-reassign
      prevArray[rowIndex] = event.target.checked;
      return (prevArray);
    });
  };

  const CheckboxRenderer = (params) => {
    // eslint-disable-next-line no-shadow
    const { rowIndex, checkedIndexes, hoveredIndex } = params;
    if (checkedIndexes[rowIndex] || hoveredIndex === rowIndex) {
      return (
        <Checkbox
          checked={checkedIndexes[rowIndex]}
          id={rowIndex}
          className={classes.checkbox}
          name={rowIndex}
          size="small"
          style={{
            display: 'flex',
            justifyContent: 'center',
            flexDirection: 'row',
            paddingTop: '0',
            paddingLeft: '0.5',
          }}
          onClick={onClickHandler}
          color="primary"
        />
      );
    }
    return (
      <Typography
        variant="body1"
        className={classes.indexStyle}
      >
        { rowIndex + 1 }
      </Typography>
    );
  };

  // eslint-disable-next-line no-unused-vars
  const [gridApi, setGridApi] = useState(null);
  // eslint-disable-next-line no-unused-vars
  const [gridColumnApi, setGridColumnApi] = useState(null);

  const onGridReady = (params) => {
    setGridApi(params.api);
  };

  const TextStyle = ({ value }) => (
    <Typography
      data-testid="text"
      className={classes.textStyle}
      variant="body1"
    >
      {value}
    </Typography>
  );

  const onCellMouseOver = ({ rowIndex }) => {
    sethoveredIndex(rowIndex);
  };

  return (

    <div
      className="ag-theme-balham"
      style={{ width: '605px' }}
    >
      <AgGridReact
        rowData={rows}
        headerHeight={40}
        onGridReady={onGridReady}
        getRowHeight={getRowHeight}
        onCellMouseOver={onCellMouseOver}
        rowSelection="multiple"
        domLayout="autoHeight"
        defaultColDef={{ resizable: true }}
        frameworkComponents={{
          relationshipRenderer: RelationshipRenderer,
          checkboxRenderer: CheckboxRenderer,
          textStyle: TextStyle,
        }}
      >
        <AgGridColumn
          pinned="left"
          width={35}
          headerCheckboxSelection
          cellRenderer="checkboxRenderer"
          cellRendererParams={{ checkedIndexes, hoveredIndex }}
          cellStyle={{
            fontSize: '1px', align: 'left', borderRight: '1px solid #eeeeee', textAlign: 'center', paddingTop: '10px',
          }}
        />

        <AgGridColumn
          field="domain"
          colId="domain"
          height={36}
          width={210}
          cellStyle={{
            align: 'center',
            paddingTop: '5px',
            justifyContent: 'center',
            borderRight: '2px solid #cccccc',
          }}
          cellRenderer="textStyle"
          headerComponentFramework={() => (
            <TableHeader
              icon1={TitleIcon}
              icon2={MoreVertIcon}
              text="Domain"
            />
          )}
        />
        <AgGridColumn
          field="trustScore"
          colId="trustScore"
          width={167}
          cellStyle={{
            align: 'center',
            paddingTop: '5px',
            justifyContent: 'center',
            textAlign: 'right',
            borderRight: '1px solid #eeeeee',
          }}
          cellRenderer="textStyle"
          headerComponentFramework={() => (
            <TableHeader
              icon1={TitleIcon}
              icon2={MoreVertIcon}
              text="TrustScore"
            />
          )}
        />

        <AgGridColumn
          field="relationship"
          colId="relationship"
          width={190}
          cellStyle={{
            align: 'center',
            paddingTop: '5px',
          }}
          cellRenderer="relationshipRenderer"
          headerComponentFramework={() => (
            <TableHeader
              icon1={CheckBoxSharpIcon}
              icon2={MoreVertIcon}
              text="Relationship"
            />
          )}
        />
      </AgGridReact>
      <RowAdder text="New Row" />
    </div>
  );
}

export default Table;
