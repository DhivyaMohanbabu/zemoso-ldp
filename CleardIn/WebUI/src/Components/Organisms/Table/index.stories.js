/* eslint-disable max-len */
import React from 'react';
import Table from '.';

export default {
  title: 'Organism/Table',
};

const Template = (args) => <Table {...args} />;

export const Default = Template.bind({});
Default.args = {
  rows: [
    {
      id: 1, domain: 'zemosolabs.com', trustScore: 22, relationship: 'vendor', checked: false,
    },
    {
      id: 2, domain: 'zemosolabs.com', trustScore: 22, relationship: 'customer', checked: false,
    },
    {
      id: 3, domain: 'zemosolabs.com', trustScore: 22, relationship: 'partner', checked: false,
    },
  ],
};
