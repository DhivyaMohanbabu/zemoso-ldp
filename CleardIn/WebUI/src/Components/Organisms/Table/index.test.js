import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { AgGridReact } from 'ag-grid-react';
import { shallow } from 'enzyme';
import Table from '.';
import theme from '../../../theme';

const rows = [
  {
    id: 1, domain: 'zemosolabs.com', trustScore: 22, relationship: 'vendor', checked: false,
  },
  {
    id: 2, domain: 'zemosolabs.com', trustScore: 22, relationship: 'customer', checked: false,
  },
  {
    id: 3, domain: 'zemosolabs.com', trustScore: 22, relationship: 'partner', checked: false,
  },
];

describe('adding rows', () => {
  it('Renders correctly with props', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Table rows={rows} />
      </ThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('first row should have expected grid data', () => {
    const component = shallow(
      <Table rows={rows} />,
    );
    expect(component).not.toBeNull();
    const agGridReactObj = component.find(AgGridReact);
    agGridReactObj.simulate('onGridReady', jest.fn());
    agGridReactObj.simulate('rowClicked', jest.fn());
    agGridReactObj.simulate('firstDataRendered', jest.fn());
    expect(agGridReactObj).toBeTruthy();
  });
}); // describe end
