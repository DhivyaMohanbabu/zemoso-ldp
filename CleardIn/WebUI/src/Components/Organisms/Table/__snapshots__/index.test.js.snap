// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adding rows Renders correctly with props 1`] = `
<div>
  <div
    class="ag-theme-balham"
    style="width: 605px;"
  >
    <div
      style="height: 100%;"
    >
      <div
        class="ag-root-wrapper ag-layout-auto-height ag-ltr"
        ref="eRootWrapper"
      >
        
                
                
        <div
          class="ag-root-wrapper-body ag-layout-auto-height ag-focus-managed"
          ref="rootWrapperBody"
        >
          <div
            class="ag-tab-guard ag-tab-guard-top"
            role="presentation"
            tabindex="0"
          />
          
                    
          <!--AG-GRID-COMP-->
          <div
            aria-colcount="4"
            aria-multiselectable="true"
            aria-rowcount="4"
            class="ag-root ag-unselectable ag-layout-auto-height"
            ref="gridPanel"
            role="grid"
            unselectable="on"
          >
            
        
            <!--AG-HEADER-ROOT-->
            <div
              class="ag-header ag-focus-managed ag-pivot-off"
              ref="headerRoot"
              role="presentation"
              style="height: 41px; min-height: 41px;"
              unselectable="on"
            >
              
            
              <div
                class="ag-pinned-left-header"
                ref="ePinnedLeftHeader"
                role="presentation"
                style="width: 35px; max-width: 35px; min-width: 35px;"
              >
                <div
                  aria-rowindex="1"
                  class="ag-header-row ag-header-row-column"
                  role="row"
                  style="top: 0px; height: 40px; width: 35px;"
                >
                  <div
                    aria-colindex="1"
                    class="ag-header-cell ag-focus-managed"
                    col-id="0"
                    role="columnheader"
                    style="width: 35px; left: 0px;"
                    tabindex="-1"
                    unselectable="on"
                  >
                    
            
                    <div
                      class="ag-header-cell-resize"
                      ref="eResize"
                      role="presentation"
                    />
                    
            
                    <!--AG-CHECKBOX-->
                    <div
                      class="ag-header-select-all ag-labeled ag-label-align-right ag-checkbox ag-input-field"
                      ref="cbSelectAll"
                      role="presentation"
                    >
                      
                
                      <div
                        class="ag-input-field-label ag-label ag-hidden ag-checkbox-label"
                        ref="eLabel"
                        role="presentation"
                      />
                      
                
                      <div
                        class="ag-wrapper ag-input-wrapper ag-checkbox-input-wrapper"
                        ref="eWrapper"
                        role="presentation"
                      >
                        
                    
                        <input
                          aria-label="Press Space to toggle all rows selection (unchecked)"
                          class="ag-input-field-input ag-checkbox-input"
                          id="ag-15-input"
                          ref="eInput"
                          tabindex="-1"
                          type="checkbox"
                        />
                        
                
                      </div>
                      
            
                    </div>
                    
        
                    <div
                      class="ag-cell-label-container"
                      role="presentation"
                    >
                      
            
                      
            
                      <div
                        class="ag-header-cell-label"
                        ref="eLabel"
                        role="presentation"
                        unselectable="on"
                      >
                        
                
                        <span
                          class="ag-header-cell-text"
                          ref="eText"
                          unselectable="on"
                        />
                        
                
                        <span
                          aria-hidden="true"
                          class="ag-header-icon ag-header-label-icon ag-filter-icon ag-hidden"
                          ref="eFilter"
                        >
                          <span
                            class="ag-icon ag-icon-filter"
                            role="presentation"
                            unselectable="on"
                          />
                        </span>
                        
                
                        
                
                        
                
                        
                
                        
            
                      </div>
                      
        
                    </div>
                  </div>
                </div>
              </div>
              
            
              <div
                class="ag-header-viewport"
                ref="eHeaderViewport"
                role="presentation"
              >
                
                
                <div
                  class="ag-header-container"
                  ref="eHeaderContainer"
                  role="rowgroup"
                  style="width: 567px;"
                >
                  <div
                    aria-rowindex="1"
                    class="ag-header-row ag-header-row-column"
                    role="row"
                    style="top: 0px; height: 40px; width: 567px;"
                  >
                    <div
                      aria-colindex="2"
                      class="ag-header-cell ag-focus-managed"
                      col-id="domain"
                      role="columnheader"
                      style="width: 210px; left: 0px;"
                      tabindex="-1"
                      unselectable="on"
                    >
                      
            
                      <div
                        class="ag-header-cell-resize"
                        ref="eResize"
                        role="presentation"
                      />
                      
            
                      <!--AG-CHECKBOX-->
                      <div
                        class="ag-header-select-all ag-labeled ag-label-align-right ag-checkbox ag-input-field ag-hidden"
                        ref="cbSelectAll"
                        role="presentation"
                      >
                        
                
                        <div
                          class="ag-input-field-label ag-label ag-hidden ag-checkbox-label"
                          ref="eLabel"
                          role="presentation"
                        />
                        
                
                        <div
                          class="ag-wrapper ag-input-wrapper ag-checkbox-input-wrapper"
                          ref="eWrapper"
                          role="presentation"
                        >
                          
                    
                          <input
                            aria-label="Press Space to toggle all rows selection (unchecked)"
                            class="ag-input-field-input ag-checkbox-input"
                            id="ag-9-input"
                            ref="eInput"
                            tabindex="-1"
                            type="checkbox"
                          />
                          
                
                        </div>
                        
            
                      </div>
                      
        
                    </div>
                    <div
                      aria-colindex="3"
                      class="ag-header-cell ag-focus-managed"
                      col-id="trustScore"
                      role="columnheader"
                      style="width: 167px; left: 210px;"
                      tabindex="-1"
                      unselectable="on"
                    >
                      
            
                      <div
                        class="ag-header-cell-resize"
                        ref="eResize"
                        role="presentation"
                      />
                      
            
                      <!--AG-CHECKBOX-->
                      <div
                        class="ag-header-select-all ag-labeled ag-label-align-right ag-checkbox ag-input-field ag-hidden"
                        ref="cbSelectAll"
                        role="presentation"
                      >
                        
                
                        <div
                          class="ag-input-field-label ag-label ag-hidden ag-checkbox-label"
                          ref="eLabel"
                          role="presentation"
                        />
                        
                
                        <div
                          class="ag-wrapper ag-input-wrapper ag-checkbox-input-wrapper"
                          ref="eWrapper"
                          role="presentation"
                        >
                          
                    
                          <input
                            aria-label="Press Space to toggle all rows selection (unchecked)"
                            class="ag-input-field-input ag-checkbox-input"
                            id="ag-11-input"
                            ref="eInput"
                            tabindex="-1"
                            type="checkbox"
                          />
                          
                
                        </div>
                        
            
                      </div>
                      
        
                    </div>
                    <div
                      aria-colindex="4"
                      class="ag-header-cell ag-focus-managed"
                      col-id="relationship"
                      role="columnheader"
                      style="width: 190px; left: 377px;"
                      tabindex="-1"
                      unselectable="on"
                    >
                      
            
                      <div
                        class="ag-header-cell-resize"
                        ref="eResize"
                        role="presentation"
                      />
                      
            
                      <!--AG-CHECKBOX-->
                      <div
                        class="ag-header-select-all ag-labeled ag-label-align-right ag-checkbox ag-input-field ag-hidden"
                        ref="cbSelectAll"
                        role="presentation"
                      >
                        
                
                        <div
                          class="ag-input-field-label ag-label ag-hidden ag-checkbox-label"
                          ref="eLabel"
                          role="presentation"
                        />
                        
                
                        <div
                          class="ag-wrapper ag-input-wrapper ag-checkbox-input-wrapper"
                          ref="eWrapper"
                          role="presentation"
                        >
                          
                    
                          <input
                            aria-label="Press Space to toggle all rows selection (unchecked)"
                            class="ag-input-field-input ag-checkbox-input"
                            id="ag-13-input"
                            ref="eInput"
                            tabindex="-1"
                            type="checkbox"
                          />
                          
                
                        </div>
                        
            
                      </div>
                      
        
                    </div>
                  </div>
                </div>
                
            
              </div>
              
            
              <div
                class="ag-pinned-right-header ag-hidden"
                ref="ePinnedRightHeader"
                role="presentation"
                style="width: 0px; max-width: 0px; min-width: 0px;"
              >
                <div
                  aria-rowindex="1"
                  class="ag-header-row ag-header-row-column"
                  role="row"
                  style="top: 0px; height: 40px; width: 0px;"
                />
              </div>
              
        
            </div>
            
        
            <div
              class="ag-floating-top"
              ref="eTop"
              role="presentation"
              style="min-height: 0px; height: 0px; display: none; overflow-y: hidden;"
              unselectable="on"
            >
              
            
              <div
                class="ag-pinned-left-floating-top"
                ref="eLeftTop"
                role="presentation"
                style="width: 35px; max-width: 35px; min-width: 35px;"
                unselectable="on"
              />
              
            
              <div
                class="ag-floating-top-viewport"
                ref="eTopViewport"
                role="presentation"
                unselectable="on"
              >
                
                
                <div
                  class="ag-floating-top-container"
                  ref="eTopContainer"
                  role="presentation"
                  style="width: 567px;"
                  unselectable="on"
                />
                
            
              </div>
              
            
              <div
                class="ag-pinned-right-floating-top ag-hidden"
                ref="eRightTop"
                role="presentation"
                unselectable="on"
              />
              
            
              <div
                class="ag-floating-top-full-width-container ag-hidden"
                ref="eTopFullWidthContainer"
                role="presentation"
                unselectable="on"
              />
              
        
            </div>
            
        
            <div
              class="ag-body-viewport ag-layout-auto-height ag-row-no-animation"
              ref="eBodyViewport"
              role="presentation"
            >
              
            
              <div
                class="ag-pinned-left-cols-container"
                ref="eLeftContainer"
                role="presentation"
                style="height: 135px; width: 35px; max-width: 35px; min-width: 35px;"
                unselectable="on"
              >
                <div
                  aria-label="Press SPACE to select this row."
                  aria-rowindex="2"
                  aria-selected="false"
                  class="ag-row ag-row-no-focus ag-row-even ag-row-level-0 ag-row-position-absolute ag-row-first"
                  comp-id="17"
                  role="row"
                  row-id="0"
                  row-index="0"
                  style="height: 45px; transform: translateY(0px); "
                >
                  <div
                    aria-colindex="1"
                    class="ag-cell ag-cell-not-inline-editing ag-cell-auto-height ag-cell-last-left-pinned ag-cell-value"
                    col-id="0"
                    comp-id="21"
                    role="gridcell"
                    style="width: 35px; left: 0px; font-size: 1px; align: left; border-right: 1px solid #eeeeee; text-align: center; padding-top: 10px; "
                    tabindex="-1"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-label="Press SPACE to select this row."
                  aria-rowindex="3"
                  aria-selected="false"
                  class="ag-row ag-row-no-focus ag-row-odd ag-row-level-0 ag-row-position-absolute"
                  comp-id="22"
                  role="row"
                  row-id="1"
                  row-index="1"
                  style="height: 45px; transform: translateY(45px); "
                >
                  <div
                    aria-colindex="1"
                    class="ag-cell ag-cell-not-inline-editing ag-cell-auto-height ag-cell-last-left-pinned ag-cell-value"
                    col-id="0"
                    comp-id="26"
                    role="gridcell"
                    style="width: 35px; left: 0px; font-size: 1px; align: left; border-right: 1px solid #eeeeee; text-align: center; padding-top: 10px; "
                    tabindex="-1"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-label="Press SPACE to select this row."
                  aria-rowindex="4"
                  aria-selected="false"
                  class="ag-row ag-row-no-focus ag-row-even ag-row-level-0 ag-row-position-absolute ag-row-last"
                  comp-id="27"
                  role="row"
                  row-id="2"
                  row-index="2"
                  style="height: 45px; transform: translateY(90px); "
                >
                  <div
                    aria-colindex="1"
                    class="ag-cell ag-cell-not-inline-editing ag-cell-auto-height ag-cell-last-left-pinned ag-cell-value"
                    col-id="0"
                    comp-id="31"
                    role="gridcell"
                    style="width: 35px; left: 0px; font-size: 1px; align: left; border-right: 1px solid #eeeeee; text-align: center; padding-top: 10px; "
                    tabindex="-1"
                    unselectable="on"
                  />
                </div>
              </div>
              
            
              <div
                class="ag-center-cols-clipper"
                ref="eCenterColsClipper"
                role="presentation"
                style="height: 135px;"
                unselectable="on"
              >
                
                
                <div
                  class="ag-center-cols-viewport"
                  ref="eCenterViewport"
                  role="presentation"
                  style="height: calc(100% + 0px);"
                >
                  
                    
                  <div
                    class="ag-center-cols-container"
                    ref="eCenterContainer"
                    role="rowgroup"
                    style="width: 567px; height: 135px;"
                    unselectable="on"
                  >
                    <div
                      aria-label="Press SPACE to select this row."
                      aria-rowindex="2"
                      aria-selected="false"
                      class="ag-row ag-row-no-focus ag-row-even ag-row-level-0 ag-row-position-absolute ag-row-first"
                      comp-id="17"
                      role="row"
                      row-id="0"
                      row-index="0"
                      style="height: 45px; transform: translateY(0px); "
                    >
                      <div
                        aria-colindex="2"
                        class="ag-cell ag-cell-not-inline-editing ag-cell-auto-height ag-cell-value"
                        col-id="domain"
                        comp-id="18"
                        role="gridcell"
                        style="width: 210px; left: 0px; align: center; padding-top: 5px; justify-content: center; border-right: 2px solid #cccccc; "
                        tabindex="-1"
                        unselectable="on"
                      />
                      <div
                        aria-colindex="3"
                        class="ag-cell ag-cell-not-inline-editing ag-cell-auto-height ag-cell-value"
                        col-id="trustScore"
                        comp-id="19"
                        role="gridcell"
                        style="width: 167px; left: 210px; align: center; padding-top: 5px; justify-content: center; text-align: right; border-right: 1px solid #eeeeee; "
                        tabindex="-1"
                        unselectable="on"
                      />
                      <div
                        aria-colindex="4"
                        class="ag-cell ag-cell-not-inline-editing ag-cell-auto-height ag-cell-value"
                        col-id="relationship"
                        comp-id="20"
                        role="gridcell"
                        style="width: 190px; left: 377px; align: center; padding-top: 5px; "
                        tabindex="-1"
                        unselectable="on"
                      />
                    </div>
                    <div
                      aria-label="Press SPACE to select this row."
                      aria-rowindex="3"
                      aria-selected="false"
                      class="ag-row ag-row-no-focus ag-row-odd ag-row-level-0 ag-row-position-absolute"
                      comp-id="22"
                      role="row"
                      row-id="1"
                      row-index="1"
                      style="height: 45px; transform: translateY(45px); "
                    >
                      <div
                        aria-colindex="2"
                        class="ag-cell ag-cell-not-inline-editing ag-cell-auto-height ag-cell-value"
                        col-id="domain"
                        comp-id="23"
                        role="gridcell"
                        style="width: 210px; left: 0px; align: center; padding-top: 5px; justify-content: center; border-right: 2px solid #cccccc; "
                        tabindex="-1"
                        unselectable="on"
                      />
                      <div
                        aria-colindex="3"
                        class="ag-cell ag-cell-not-inline-editing ag-cell-auto-height ag-cell-value"
                        col-id="trustScore"
                        comp-id="24"
                        role="gridcell"
                        style="width: 167px; left: 210px; align: center; padding-top: 5px; justify-content: center; text-align: right; border-right: 1px solid #eeeeee; "
                        tabindex="-1"
                        unselectable="on"
                      />
                      <div
                        aria-colindex="4"
                        class="ag-cell ag-cell-not-inline-editing ag-cell-auto-height ag-cell-value"
                        col-id="relationship"
                        comp-id="25"
                        role="gridcell"
                        style="width: 190px; left: 377px; align: center; padding-top: 5px; "
                        tabindex="-1"
                        unselectable="on"
                      />
                    </div>
                    <div
                      aria-label="Press SPACE to select this row."
                      aria-rowindex="4"
                      aria-selected="false"
                      class="ag-row ag-row-no-focus ag-row-even ag-row-level-0 ag-row-position-absolute ag-row-last"
                      comp-id="27"
                      role="row"
                      row-id="2"
                      row-index="2"
                      style="height: 45px; transform: translateY(90px); "
                    >
                      <div
                        aria-colindex="2"
                        class="ag-cell ag-cell-not-inline-editing ag-cell-auto-height ag-cell-value"
                        col-id="domain"
                        comp-id="28"
                        role="gridcell"
                        style="width: 210px; left: 0px; align: center; padding-top: 5px; justify-content: center; border-right: 2px solid #cccccc; "
                        tabindex="-1"
                        unselectable="on"
                      />
                      <div
                        aria-colindex="3"
                        class="ag-cell ag-cell-not-inline-editing ag-cell-auto-height ag-cell-value"
                        col-id="trustScore"
                        comp-id="29"
                        role="gridcell"
                        style="width: 167px; left: 210px; align: center; padding-top: 5px; justify-content: center; text-align: right; border-right: 1px solid #eeeeee; "
                        tabindex="-1"
                        unselectable="on"
                      />
                      <div
                        aria-colindex="4"
                        class="ag-cell ag-cell-not-inline-editing ag-cell-auto-height ag-cell-value"
                        col-id="relationship"
                        comp-id="30"
                        role="gridcell"
                        style="width: 190px; left: 377px; align: center; padding-top: 5px; "
                        tabindex="-1"
                        unselectable="on"
                      />
                    </div>
                  </div>
                  
                
                </div>
                
            
              </div>
              
            
              <div
                class="ag-pinned-right-cols-container ag-hidden"
                ref="eRightContainer"
                role="presentation"
                style="height: 135px;"
                unselectable="on"
              >
                <div
                  aria-label="Press SPACE to select this row."
                  aria-rowindex="2"
                  aria-selected="false"
                  class="ag-row ag-row-no-focus ag-row-even ag-row-level-0 ag-row-position-absolute ag-row-first"
                  comp-id="17"
                  role="row"
                  row-id="0"
                  row-index="0"
                  style="height: 45px; transform: translateY(0px); "
                />
                <div
                  aria-label="Press SPACE to select this row."
                  aria-rowindex="3"
                  aria-selected="false"
                  class="ag-row ag-row-no-focus ag-row-odd ag-row-level-0 ag-row-position-absolute"
                  comp-id="22"
                  role="row"
                  row-id="1"
                  row-index="1"
                  style="height: 45px; transform: translateY(45px); "
                />
                <div
                  aria-label="Press SPACE to select this row."
                  aria-rowindex="4"
                  aria-selected="false"
                  class="ag-row ag-row-no-focus ag-row-even ag-row-level-0 ag-row-position-absolute ag-row-last"
                  comp-id="27"
                  role="row"
                  row-id="2"
                  row-index="2"
                  style="height: 45px; transform: translateY(90px); "
                />
              </div>
              
            
              <div
                class="ag-full-width-container"
                ref="eFullWidthContainer"
                role="presentation"
                style="height: 135px;"
                unselectable="on"
              />
              
        
            </div>
            
        
            <div
              class="ag-floating-bottom"
              ref="eBottom"
              role="presentation"
              style="min-height: 0px; height: 0px; display: none; overflow-y: hidden;"
              unselectable="on"
            >
              
            
              <div
                class="ag-pinned-left-floating-bottom"
                ref="eLeftBottom"
                role="presentation"
                style="width: 35px; max-width: 35px; min-width: 35px;"
                unselectable="on"
              />
              
            
              <div
                class="ag-floating-bottom-viewport"
                ref="eBottomViewport"
                role="presentation"
                unselectable="on"
              >
                
                
                <div
                  class="ag-floating-bottom-container"
                  ref="eBottomContainer"
                  role="presentation"
                  style="width: 567px;"
                  unselectable="on"
                />
                
            
              </div>
              
            
              <div
                class="ag-pinned-right-floating-bottom ag-hidden"
                ref="eRightBottom"
                role="presentation"
                unselectable="on"
              />
              
            
              <div
                class="ag-floating-bottom-full-width-container ag-hidden"
                ref="eBottomFullWidthContainer"
                role="presentation"
                unselectable="on"
              />
              
        
            </div>
            
        
            <div
              aria-hidden="true"
              class="ag-body-horizontal-scroll"
              ref="eHorizontalScrollBody"
              style="height: 0px; max-height: 0px; min-height: 0px;"
            >
              
            
              <div
                class="ag-horizontal-left-spacer"
                ref="eHorizontalLeftSpacer"
                style="width: 35px; max-width: 35px; min-width: 35px;"
              />
              
            
              <div
                class="ag-body-horizontal-scroll-viewport"
                ref="eBodyHorizontalScrollViewport"
                style="height: 0px; max-height: 0px; min-height: 0px;"
              >
                
                
                <div
                  class="ag-body-horizontal-scroll-container"
                  ref="eBodyHorizontalScrollContainer"
                  style="width: 567px; height: 0px; max-height: 0px; min-height: 0px;"
                />
                
            
              </div>
              
            
              <div
                class="ag-horizontal-right-spacer"
                ref="eHorizontalRightSpacer"
                style="width: 0px; max-width: 0px; min-width: 0px;"
              />
              
        
            </div>
            
        
            <!--AG-OVERLAY-WRAPPER-->
            <div
              aria-hidden="true"
              class="ag-overlay ag-hidden"
              ref="overlayWrapper"
            >
              
            
              <div
                class="ag-overlay-panel"
              >
                
                
                <div
                  class="ag-overlay-wrapper ag-layout-auto-height"
                  ref="eOverlayWrapper"
                />
                
            
              </div>
              
        
            </div>
            
    
          </div>
          
                    
                
          <div
            class="ag-tab-guard ag-tab-guard-bottom"
            role="presentation"
            tabindex="0"
          />
        </div>
        
                
                
        <!--AG-PAGINATION-->
        <div
          aria-describedby="ag-4-start-page ag-4-start-page-number ag-4-of-page ag-4-of-page-number ag-4-first-row ag-4-to ag-4-last-row ag-4-of ag-4-row-count"
          aria-live="polite"
          class="ag-paging-panel ag-unselectable ag-hidden"
          id="ag-4"
        >
          
                
          <span
            aria-hidden="true"
            class="ag-paging-row-summary-panel"
          >
            
                    
            <span
              class="ag-paging-row-summary-panel-number"
              id="ag-4-first-row"
              ref="lbFirstRowOnPage"
            />
            
                    
            <span
              id="ag-4-to"
            >
              to
            </span>
            
                    
            <span
              class="ag-paging-row-summary-panel-number"
              id="ag-4-last-row"
              ref="lbLastRowOnPage"
            />
            
                    
            <span
              id="ag-4-of"
            >
              of
            </span>
            
                    
            <span
              class="ag-paging-row-summary-panel-number"
              id="ag-4-row-count"
              ref="lbRecordCount"
            />
            
                
          </span>
          
                
          <span
            class="ag-paging-page-summary-panel"
            role="presentation"
          >
            
                    
            <div
              aria-label="First Page"
              class="ag-paging-button"
              ref="btFirst"
              role="button"
              tabindex="0"
            >
              <span
                class="ag-icon ag-icon-first"
                role="presentation"
                unselectable="on"
              />
            </div>
            
                    
            <div
              aria-label="Previous Page"
              class="ag-paging-button"
              ref="btPrevious"
              role="button"
              tabindex="0"
            >
              <span
                class="ag-icon ag-icon-previous"
                role="presentation"
                unselectable="on"
              />
            </div>
            
                    
            <span
              aria-hidden="true"
              class="ag-paging-description"
            >
              
                        
              <span
                id="ag-4-start-page"
              >
                Page
              </span>
              
                        
              <span
                class="ag-paging-number"
                id="ag-4-start-page-number"
                ref="lbCurrent"
              />
              
                        
              <span
                id="ag-4-of-page"
              >
                of
              </span>
              
                        
              <span
                class="ag-paging-number"
                id="ag-4-of-page-number"
                ref="lbTotal"
              />
              
                    
            </span>
            
                    
            <div
              aria-label="Next Page"
              class="ag-paging-button"
              ref="btNext"
              role="button"
              tabindex="0"
            >
              <span
                class="ag-icon ag-icon-next"
                role="presentation"
                unselectable="on"
              />
            </div>
            
                    
            <div
              aria-label="Last Page"
              class="ag-paging-button"
              ref="btLast"
              role="button"
              tabindex="0"
            >
              <span
                class="ag-icon ag-icon-last"
                role="presentation"
                unselectable="on"
              />
            </div>
            
                
          </span>
          
            
        </div>
        
                
            
      </div>
    </div>
    <div
      class="makeStyles-rowAdder-4"
      role="button"
      tabindex="0"
    >
      <div
        class="makeStyles-rowAdderBox-5"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
          />
        </svg>
        <p
          class="MuiTypography-root makeStyles-rowAdderText-6 MuiTypography-body1 MuiTypography-noWrap"
          data-testid="text"
        >
          New Row
        </p>
      </div>
    </div>
  </div>
</div>
`;
